types/lodash/lodash-tests.ts

```diff
@@ -11556,65 +11693,332 @@ namespace TestMapKeys {
+        _(numericDictionary).mapKeys(numericDictionaryIterator); // $ExpectType LoDashImplicitWrapper<Dictionary<AbcObject>>
+        _(numericDictionary).mapKeys(''); // $ExpectType LoDashImplicitWrapper<Dictionary<AbcObject>>
+        _(numericDictionary).mapKeys({}); // $ExpectType LoDashImplicitWrapper<Dictionary<AbcObject>>
     }
 
     {
-        let result: _.LoDashExplicitObjectWrapper<_.Dictionary<AbcObject>>;
+        _(abcObject).mapKeys(); // $ExpectType LoDashImplicitWrapper<Dictionary<string | number | boolean>>
+        _(abcObject).mapKeys(abcObjectIterator); // $ExpectType LoDashImplicitWrapper<Dictionary<string | number | boolean>>
+        _(abcObject).mapKeys(''); // $ExpectType LoDashImplicitWrapper<Dictionary<string | number | boolean>>
+    }
+
+    {
+        _(array).chain().mapKeys(); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(array).chain().mapKeys(listIterator); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(array).chain().mapKeys(''); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(array).chain().mapKeys({}); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+
+        _(list).chain().mapKeys(); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(list).chain().mapKeys(listIterator); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(list).chain().mapKeys(''); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(list).chain().mapKeys({}); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+
+        _(dictionary).chain().mapKeys(); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(dictionary).chain().mapKeys(dictionaryIterator); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(dictionary).chain().mapKeys(''); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(dictionary).chain().mapKeys({}); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+
+        _(numericDictionary).chain().mapKeys(); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(numericDictionary).chain().mapKeys(numericDictionaryIterator); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(numericDictionary).chain().mapKeys(''); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+        _(numericDictionary).chain().mapKeys({}); // $ExpectType LoDashExplicitWrapper<Dictionary<AbcObject>>
+    }
+
+    {
+        _(abcObject).chain().mapKeys(); // $ExpectType LoDashExplicitWrapper<Dictionary<string | number | boolean>>
+        _(abcObject).chain().mapKeys(abcObjectIterator); // $ExpectType LoDashExplicitWrapper<Dictionary<string | number | boolean>>
+        _(abcObject).chain().mapKeys(''); // $ExpectType LoDashExplicitWrapper<Dictionary<string | number | boolean>>
+    }
+}
+
+// _.mapValues
```

Comment:
Wow, I didn't even realize we were missing `mapValues` tests.

Looks really good. Although probably make a few adjustments based on my previous comments.

Reply:
Also, I feel like all of the braces (`{}`) aren't necessary since you don't ever do `let result: x;`. For other tests, the braces are needed to keep scopes separate. But that's not an issue here, so I would just use line breaks instead of braces to logically group things.